evaluation.py

```diff
@@ -146,6 +147,101 @@ def _non_max_supression(self, plain, threshold, windowSize=3):
         under_thresh_indices = plain < threshold
         plain[under_thresh_indices] = 0
         return plain * (plain == maximum_filter(plain, footprint=np.ones((windowSize, windowSize))))
+        img_v_resize = self._vstack_images(heatmap_imgs)
+
+        cv2.imwrite(filename, img_v_resize)
+
+    """
+        Parameters
+        ----------
+        metadata : object
+        should be metadata associated to a single image
+
+        untransformed_x : int
+        x coordinate to
+    """
+    def __undo_x(self, metadata, untransformed_x):
+      predicted_x = round(untransformed_x * metadata['cropped_width'] / metadata['input_dim'][0] + metadata['anchor_x'])
+      return int(predicted_x)
+
+    """
+        Parameters
+        ----------
+        metadata : object
+        should be metadata associated to a single image
+
+        untransformed_y : int
+        x coordinate to
+    """
+    def __undo_y(self, metadata, untransformed_y):
+      predicted_y = round(untransformed_y * metadata['cropped_height'] / metadata['input_dim'][1] + metadata['anchor_y'])
+      return int(predicted_y)
+
+    """
+        Parameters
+        ----------
+        metadata : object
+        should be metadata associated to a single image
+
+        untransformed_predictions : list
+        a list of precitions that need to be transformed
+        Example:  [1,2,0,1,4,666,32...]
+    """
+    def undo_bounding_box_transformations(self, metadata, untransformed_predictions):
```

Comment:
just to confirm, have you validated this function against some ground truth data? (Take the transformed annotations, scatter plot on the transformed image. This is our baseline (what we feed to the model during train/val). Then take the same transformed annotations and run it through the `undo_bounding_box_transformations` and scatter plot on the original image. Maybe even plot the original annotations. They should line up).

Reply:
Yes I did have a chance to test these changes. I used ground truth keypoints that had been transformed, then I called theses functions on them to make sure I could get back to the ground truth.

Reply:
awesome, sounds good. :)